cloudplatform/cloudplatform-connectivity-scp-cf/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ScpCfDestinationServiceAdapter.java

```diff
@@ -83,16 +84,23 @@ String getProviderTenantId()
         if( providerTenantId != null ) {
             return providerTenantId;
         }
-        try {
+
+        final TypedMapView credentials = Try.of(() -> {
             final ServiceBinding binding = serviceBindingSupplier.get();
+            return TypedMapView.ofCredentials(binding);
+        })
+            .getOrElseThrow(
+                e -> new DestinationAccessException(
+                    "Could not resolve destination to Destination Service on behalf of provider.",
+                    e));
```

Comment:
[minor]
Formatting is pretty odd IMO.

How about:
```suggestion
        final Try<ServiceBinding> maybeServiceBinding = Try.ofSupplier(serviceBindingSupplier);
        if( maybeServiceBinding.isFailure() ) {
            final String msg = "Could not resolve destination to Destination Service on behalf of provider.";
            throw new DestinationAccessException(msg, maybeServiceBinding.getCause());
        }

        final TypedMapView credentials = TypedMapView.ofCredentials(maybeServiceBinding.get());
```

Reply:
TypedMapView.ofCredentials can throw many Exceptions, I thought it was safer this way.

Reply:
> `TypedMapView.ofCredentials` can throw many Exceptions

True... didn't consider that.
We could change the code to something like that:

```suggestion
        final Try<ServiceBinding> maybeServiceBinding = Try.ofSupplier(serviceBindingSupplier);
        if( maybeServiceBinding.isFailure() ) {
            final String msg = "Could not resolve destination to Destination Service on behalf of provider.";
            throw new DestinationAccessException(msg, maybeServiceBinding.getCause());
        }
        
        final Try<TypedMapView> maybeCredentials = Try.of(() -> TypedMapView.ofCredentials(maybeServiceBinding.get()));
        if (maybeCredentials.isFailure()) {
            final String msg = "Could not extract credentials from destination to Destination Service.";
            throw new DestinationAccessException(msg, maybeCredentials.getCause());
        }

        final TypedMapView credentials = maybeCredentials.get();
```

But I'm not sure whether it makes it more readable (maybe a tiny bit).